### PR TITLE
Feature/057 create separate library component with signs

### PR DESCRIPTION
**Pull request by PavelOlkhovoi**

### CODE REVIEW

File: src/components/designer/LibraryRoadSignsButton.jsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
+import { useState } from "react";
+import "./designer-style.css";
+import { Dropdown, Space } from "antd";
+import { SettingOutlined } from "@ant-design/icons";
+import { useDispatch } from "react-redux";
+// import { storeTempSignsLibMode } from "../../store/slices/application";
```
Comment (d4v3000) [inline src/components/designer/LibraryRoadSignsButton.jsx]: remove

File: src/components/designer/LibraryRoadSignsButton.jsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import { storeSignsLibMode } from "../../store/slices/signsLibrary";
+import { useSearchParams } from "react-router-dom";
+
+const LibraryRoadSignsButton = ({
+  connectionId,
+  setCurrentMode,
+  setIsdragging,
+}) => {
+  // const signLibMode = useSelector(getTempSignsLibMode);
```
Comment (d4v3000) [inline src/components/designer/LibraryRoadSignsButton.jsx]: remove

File: src/components/designer/LibraryRoadSignsButton.jsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+      key: "1",
+    },
+    {
+      label: (
+        <>
+          <div
+            onClick={() => {
+              dispatch(storeSignsLibMode("timeline"));
+              setCurrentMode("timeline");
+              // setUrlParams({ channel: connectionId, mode: "timeline" });
```
Comment (d4v3000) [inline src/components/designer/LibraryRoadSignsButton.jsx]: remove

File: src/components/designer/LibraryRoadSignsButton.jsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+      key: "2",
+    },
+    {
+      label: (
+        <>
+          <div
+            onClick={() => {
+              dispatch(storeSignsLibMode("overlay"));
+              setCurrentMode("overlay");
+              // setUrlParams({ channel: connectionId, mode: "overlay" });
```
Comment (d4v3000) [inline src/components/designer/LibraryRoadSignsButton.jsx]: remove

File: src/components/designer/LibraryRoadSignsButton.jsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+      key: "3",
+    },
+    {
+      label: (
+        <>
+          <div
+            onClick={() => {
+              dispatch(storeSignsLibMode("inside"));
+              setCurrentMode("inside");
+              // setUrlParams({ channel: connectionId, mode: "inside" });
```
Comment (d4v3000) [inline src/components/designer/LibraryRoadSignsButton.jsx]: remove

File: src/components/designer/LibraryRoadSignsButton.jsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    <Dropdown
+      menu={{
+        items,
+        onClick: handleMenuClick,
+      }}
+      onOpenChange={handleOpenChange}
+      open={open}
+      style={{ flexShrink: 0, marginLeft: "auto" }}
+    >
+      <a onClick={(e) => e.preventDefault()}>
```
Comment (d4v3000) [inline src/components/designer/LibraryRoadSignsButton.jsx:122]: What is the reason for the <a> tag here? You want to use the <a> tag as a redirection tool with the href prop and don't want to use onClick. And what does the onClick even do here?

File: src/components/designer/SignsLibrary.jsx [hunk 7]
```diff
@@ -0,0 +1,475 @@
...
+      {groupItems ? groupItems : group.iconsArr.length}
+    </span>
+  </div>
+);
+
+const SignsLibrary = ({
+  dataIn: signLibrary = signLocal,
+  extractor = libraryExtractor,
+  height = "100%",
+  // margins = "15px 0 12px 4px",
```
Comment (d4v3000) [inline src/components/designer/SignsLibrary.jsx]: remove

File: src/components/designer/SignsLibrary.jsx [hunk 8]
```diff
@@ -0,0 +1,475 @@
...
+            width,
+            border: "1px solid #F0F0F0",
+            padding: "10px 20px",
+            boxShadow: "rgba(15, 14, 15, 0.07) 4px 1px 9px 1px",
+            borderRadius: "12px",
+            overflow: "auto",
+            height,
+            background: "#ffff",
+            color: "#1b1b1f",
+            // boxSizing: "content-box",
```
Comment (d4v3000) [inline src/components/designer/SignsLibrary.jsx]: remove

File: src/components/designer/SignsLibrary.jsx [hunk 9]
```diff
@@ -0,0 +1,475 @@
...
+              <div style={{ marginLeft: "auto" }}>
+                {/* <PushpinOutlined
+                  style={{
+                    color: "#a5a5a5",
+                    fontSize: "16px",
+                    fontWeight: "bold",
+                    marginRight: "12px",
+                  }}
+                  onClick={() => setIsPinnedLibrary(false)}
+                /> */}
```
Comment (d4v3000) [inline src/components/designer/SignsLibrary.jsx:347]: remove

File: src/components/designer/SignsLibrary.jsx [hunk 10]
```diff
@@ -0,0 +1,475 @@
...
+                  onClick={() => {
+                    if (urlParams.get("channel")) {
+                      window.close();
+                    }
+                    closeCallBack
+                      ? closeCallBack()
+                      : dispatch(storeSignsLibMode("none"));
+                  }}
+                  style={{
+                    // color: colorPrimary,
```
Comment (d4v3000) [inline src/components/designer/SignsLibrary.jsx]: remove

File: src/components/designer/SignsLibrary.jsx [hunk 11]
```diff
@@ -0,0 +1,475 @@
...
+                    fontSize: "16px",
+                    fontWeight: "medium",
+                    color: "#1b1b1f",
+                  }}
+                />
+              </div>
+            </div>
+            <Divider style={{ margin: "22px 0px" }} />
+            <div style={{ margin: "15px 0px" }}>
+              {/* <span style={libraryTitle}>Suche</span> */}
```
Comment (d4v3000) [inline src/components/designer/SignsLibrary.jsx]: remove

File: src/components/designer/signs-style.js [hunk 12]
```diff
@@ -0,0 +1,47 @@
+export const colorPrimary = "#6965db";
+export const colorInactiv = "#a5a5a5";
+export const colorTextBlack = "#1b1b1f";
+export const iconWrapperSize = {
+  // width: "28px",
```
Comment (d4v3000) [inline src/components/designer/signs-style.js]: remove

File: src/components/timeline/DrawingCard.jsx [hunk 13]
```diff
@@ -5,10 +5,16 @@ import {
   deleteTimelineObject,
   getCurrentApplication,
   storeCurrentApplication,
+  // storeTempSignsLibMode,
+  // storeTempEditingDrawing,
```
Comment (d4v3000) [inline src/components/timeline/DrawingCard.jsx]: remove

File: src/store/slices/application.js [hunk 14]
```diff
@@ -186,6 +186,9 @@ export const {
   updateTimelineTitle,
   updateTimelineStatus,
   deleteTimelineObject,
+  // storeTempSignsLibMode,
+  // storeTempSignsLibIconClicked,
+  // storeTempEditingDrawing,
```
Comment (d4v3000) [inline src/store/slices/application.js]: remove

File: src/components/designer/TabsConnection.jsx [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+import { useState, useEffect } from "react";
+
+const TabsConnection = ({ channelId, addImage }) => {
+  const [message, setMessage] = useState("");
+  const [channel, setChannel] = useState(null);
+
+  useEffect(() => {
+    const newChannel = new BroadcastChannel(channelId);
+    const handleMessage = (newMessage) => {
+      console.log("Received message in TempTabsConnection", newMessage);
```
Comment (d4v3000) [inline src/components/designer/TabsConnection.jsx]: remove

**Review by d4v3000 — COMMENTED** [review]
Mostly comments to remove unused code and logs right now

File: src/components/designer/LibraryRoadSignsButton.jsx (hunk 6 above)
Comment (PavelOlkhovoi) [reply src/components/designer/LibraryRoadSignsButton.jsx:122]: I copied that code from somewhere. I deleted it and everything works fine.

File: src/components/designer/SignsLibrary.jsx (hunk 9 above)
Comment (PavelOlkhovoi) [reply src/components/designer/SignsLibrary.jsx:347]: I did

**Pull request merged**